packages/components/src/badge/stories/index.story.tsx

```diff
@@ -13,6 +13,22 @@ const meta = {
 	title: 'Components/Containers/Badge',
 	id: 'components-badge',
 	tags: [ 'status-private' ],
+	args: {
+		children: 'Code is Poetry',
+		intent: 'default',
+		// @ts-ignore
+		'...props': null,
+	},
+	argTypes: {
+		// @ts-ignore
+		'...props': {
+			control: {
+				type: null,
+			},
+			description:
+				'Any other props will be passed down to the underlying `span` element',
```

Comment:
Wouldn't this be making an internal implementation detail (the component's DOM structure) part of the documentation?

Reply:
I think it's pretty much unavoidable to document this, as long as we allow passing down rest props. It is already semi-official as part of the TypeScript types. Are you thinking of maybe using a different wording? For example, we could say something like "This component also accepts props of `span`".

Reply:
> It is already semi-official as part of the TypeScript types

True 🤔  Out of curiosity, do you think that changing such element (say, from `span` to `div`) would be considered a breaking change?

> This component also accepts props of span

That already sounds better, as it removes any hints about the actual structure.

Reply:
> do you think that changing such element (say, from `span` to `div`) would be considered a breaking change?

In most cases I'd say no, at least in terms of props/functionality.

I'm also open to reconsidering the necessity of #67979 if you're apprehensive about it.

Reply:
I think that something like "This component also accepts props of `span`" hits a good compromise between documenting rest props without exposing too much info about internal implementation details. Thank you!